Bonsai.Editor/GraphModel/WorkflowEditor.cs

```diff
@@ -1114,9 +1114,10 @@ private void ConfigureBuilder(ExpressionBuilder builder, GraphNode selectedNode,
             CreateGraphNodeType nodeType)
         {
             // Estimate number of inputs to the nested node
-            var inputCount = workflowBuilder.ArgumentRange.LowerBound;
-            if (nodeType == CreateGraphNodeType.Successor) inputCount = Math.Max(inputCount, selectedNodes.Count());
-            else inputCount = Math.Max(inputCount, selectedNodes.Sum(node => workflow.PredecessorEdges(node).Count()));
+            var inputCount = nodeType == CreateGraphNodeType.Successor
+                ? selectedNodes.Count(node => !node.Value.IsBuildDependency())
+                : selectedNodes.Sum(node => workflow.PredecessorEdges(node).Count(edge => !edge.Item1.Value.IsBuildDependency()));
```

Comment:
More for my own understanding: Can you explain `IsBuildDependency` here? I expected that the `VisualizerMapping` would *not* be a build dependency and the incoming node *would*, but looking in the debugger (and reading this code) it seems the opposite is true.

I would think `IsBuildDependency` would be true for nodes which are involved in compilation, is that not what it's for? (The documentation on the method isn't very helpful 😅)

Reply:
Yeah, `BuildDependency` might not be a great name. Basically `VisualizerMapping` and the other mapping or externalized property nodes are "build dependencies" in the sense that they must be evaluated before their target nodes, but their outputs are not passed directly as input arguments to the `Build` method of their target node.

The reason it is not the best name is that of course the other nodes are also "build dependencies" in the sense that passing arguments also implicitly requires that you build before your target. At the time we didn't want to call them "mapping" nodes because they may play other functions (e.g. externalized property is very different from the others). I think I took this slightly inspired by "development dependencies" in the old NuGet `packages.config` but the choice of name is really not great.

Would be happy to take in proposals to change the name. If you have a better suggestion feel free to open an issue and we can obsolete this one and change it for 3.0.

Reply:
Ah, I can see how the name came into being now. Not sure if a better name immediately comes to mind but I'll let you know if I think of something. It may just be enough to give visualizer mappings and externalized properties as examples in the remarks or something.